### ISSUE TITLE
Migrate from Dendrite to Synapse

### ISSUE THREAD

**Issue by bahuma20** [matrix-org/dendrite#3116]
As the Dendrite project is currently orientating for a different use case i want to switch back to Synapse as it has more features and is more stable.

Is that possible?

I have a single-user instance and i am fine with loosing all my old chats (so no migration of those needed, i will just re-join groups on synapse later)

The only thing that i want is to not lock out of my domain and keep my username.

Can i just use the same private key for synapse then? Or are there other things to consider?

**Comment by axiopaladin** [comment]
If you *don't* need to migrate data, there's no reason you can't simply deactivate your current account, stop the Dendrite software on your server, and then start up a new Synapse instance on the same machine (or a new one, if you update your domain's DNS records to point to the new machine's IP) and create a new account with the same username on the new instance.

If you *do* need to migrate data, that's discussed in #1705, albeit in the opposite direction. In either case, I don't think it's possible to migrate data at this time.

**Comment by andzejsp** [comment]
is there a way to atleast export the room list? maybe from element app, but i havent seen that option.. also can someone tell me how to add old dendrite server keys to the new matrix synapse keys. documentation is written for phd enjoyers, im just a causal, a meer mortal, i dont understand what it all means.

excerpt:

[old_signing_keys](https://matrix-org.github.io/synapse/latest/usage/configuration/config_documentation.html#old_signing_keys)

The keys that the server used to sign messages with but won't use to sign new messages. For each key, key should be the base64-encoded public key, and expired_tsshould be the time (in milliseconds since the unix epoch) that it was last used.

It is possible to build an entry from an old signing.key file using the export_signing_key script which is provided with synapse.

Example configuration:

old_signing_keys:
  "ed25519:id": { key: "base64string", expired_ts: 123456789123 }


i have the key, but where do i get the expired_ts??

**Comment by bahuma20** [comment]
> i have the key, but where do i get the expired_ts??

The documentation says `expired_ts should be the time (in milliseconds since the unix epoch) that it was last used.`. This means the time when you switched.

**Comment by bahuma20** [comment]
@axiopaladin I am a little worried about your solution.

When i want to deactivate my account via element it shows this warning:

![image](https://github.com/matrix-org/dendrite/assets/5746121/5cf4256b-63a5-4223-9158-f97b9451bc95)

And the part about "No one will be able to reuse your username" is the thing i want to prevent.

Is that somehow possible?

**Comment by axiopaladin** [comment]
@bahuma20 The homeserver is the only thing enforcing no-reuse. If you take the homeserver down completely and replace it with a fresh new one, it will not know to prevent registration with that name.

**Comment by cyberb** [comment]
I am also unlucky one to start using dendrite and have bridge bots issues, I am thinking about migrating away to conduit or synapse.
I am afraid of migrating to Synapse as I usually use underpowered devices and that is why thinking about moving to Conduit.
Not sure what happened to dendrite and why it stopped progressing and now conduit seems to be a better option.
@bahuma20 did you finish the migration? are you happy with Synapse?

**Comment by bertybuttface** [comment]
@cyberb There have been some recent changes, I am no expert but they might help you with your bridge bot issues. In particular this PR which was recently merged: https://github.com/matrix-org/dendrite/pull/3078

**Comment by cyberb** [comment]
@bertybuttface oh good to know, let me try!
I was advised to not use dendrite here https://github.com/mautrix/slack/issues/41 but it works so good (apart from this login issue) so I am willing to try any hot fix at this point.

**Comment by bertybuttface** [comment]
I think there are still a few more bits that need sorting (see kuhnchris comment at bottom of that issue) for bridge support to work perfectly. You might be better off following the advice you were provided with, I'm definitely no expert.

**Comment by S7evinK** [comment]
FYI: https://github.com/mautrix/slack/issues/41 is most likely a race condition between the syncapi (handling `/joined_members`) and the appservice. (and probably explains more of the bridge issues)
If the AS handles the event before the syncapi had a chance, the bridge will be sad that there are no users and issues a `/leave`

**Comment by cyberb** [comment]
FYI: good news, this seems to have fixed all my go bridge bot invite issues: https://github.com/matrix-org/dendrite/pull/3078

**Comment by bahuma20** [comment]
@cyberb Yes i have disabled my old account on Dendrite via Element (Disable account in the settings). That forced the account to leave all rooms i was in.

Then i setup Synapse and created a new account with the same name. 

Everything worked
